### DISCUSSION TITLE
syntax question

### DISCUSSION THREAD

**Discussion by holybtn** [mickem/nscp#838] [General]
Hello,
First use of nsclient on a windows for checking some specific services.

So from the nagios i can do this
./check_nrpe -2 -H server -c check_service
CRITICAL: GoogleUpdaterInternalService132.0.6806.0=stopped (auto), GoogleUpdaterService132.0.6806.0=stopped (auto), RemoteRegistry=stopped (auto), WbioSrvc=stopped (auto), delayed (BITS=stopped (delayed), edgeupdate=stopped (delayed), gupdate=stopped (delayed), MapsBroker=stopped (delayed), sppsvc=stopped (delayed))

from this i have no idea how to check for a speciofic service

I tried
./check_nrpe -2 -H server -c check_service service=snmp

doesn't get something relevant, am i missing something?

tx

**Comment by mickem** [comment]
IIRC check_nrpe needs a -a before the arguments.
There is also some config required to allow arguments from NRPE: https://nsclient.org/docs/howto/nrpe/#nsclient-configuration-revisited

But your assumption works:

Running from NSClient++ test client.
![image](https://github.com/user-attachments/assets/7c67c38a-0ae6-4dee-b89e-5d7d2333ce58)

**Comment by holybtn** [comment]
Hello ty for your comment

From the host i have like you so it works
It's from the linux that i can't have that.

so i can do this
./check_nrpe -2 -H server -c check_service

but i can't specify a service name

the -a you talk about where do i put this?

tx

**Comment by mickem** [comment]
``` 
./check_nrpe -2 -H server -c check_service -a service=LanmanServer
```

**Comment by holybtn** [comment]
Hello
It doesn't work, no idea what i am missing


 ./check_nrpe -2 -H srv -c check_service -a service=LanmanServer
Exception processing request: Request contained arguments (not currently allowed, check the allow arguments option).

**Comment by mickem** [comment]
NSClient++ is not configured to allow arguments if you check the docs I linked: https://nsclient.org/docs/howto/nrpe/#nsclient-configuration-revisited

![image](https://github.com/user-attachments/assets/724909a1-4b81-4f4b-8dd6-f8f751f1b531)

To better understand the security implications of the various argument configurations I created this page: https://nsclient.org/docs/howto/external_scripts/#arguments

**Comment by holybtn** [comment]
Thanks a lot, you saved it!

Tx again
